google/datalab/utils/commands/_utils.py

```diff
@@ -336,6 +338,62 @@ def parse_config(config, env, as_dict=True):
   return config
 
 
+def parse_config_for_selected_keys(content, env, keys):
+  """ Parse a config from a magic cell body for selected config keys.
+
+  For example, if 'content' is:
+    config_item1: value1
+    config_item2: value2
+    config_item3: value3
+  and 'keys' are: [config_item1, config_item3]
+
+  The results will be a tuple of
+  1. The parsed config items (dict): {config_item1: value1, config_item3: value3}
+  2. The remaining content (string): config_item2: value2
+
+  Args:
+    content: the input content. A string. It has to be a yaml or JSON string.
+    env: user supplied dictionary for replacing vars (for example, $myvalue).
+    keys: a list of keys to retrieve from content.
```

Comment:
add:

keys must be a top-level key. So if content is  
config_item4:
   config_item5: True

only config_item4 could be a key